doc/guides/bluetooth/bluetooth-arch.rst

```diff
@@ -0,0 +1,197 @@
+  low-level, real-time protocol which provides, in conjunction with the Radio
+  Hardware, standard interoperable over the air communication. The LL schedules
+  packet reception and transmission, guarantees the delivery of data, and
+  handles all the LL control procedures
+* **Radio Hardware**: The required analog and digital baseband functional blocks
+  that permit the Link Layer firmware to send and receive in the 2.4GHz band of
+  the spectrum
+
+.. _bluetooth-hci:
+
+Host Controller Interface
+=========================
+
+The Bluetooth Specification describes the format in which a Host must
+communicate with a Controller. This is called the Host Controller Interface
+(HCI) protocol. HCI can be implemented over a range of different physical
+transports like UART, SPI or USB. This protocol defines the commands that a Host
+can send to a Controller and the events that it can expect in return, and also
+the format for user and protocol data that needs to go over the air. The HCI
+ensures that different Host and Controller implementations can communicate
+in a standard way making it possible to combine Hosts and Controllers from
+different vendors.
+
+.. _bluetooth-configs:
+
+Configurations
+==============
+
+The three separate layers of the protocol and the standardized interface make
+it possible to implement the Host and Controller on different platforms. The two
+following typical configurations are commonly used:
+
+* **Single-chip configuration**: In this configuration, a single microcontroller
+  contains all three layers and the application itself. This can also be called a
+  system-on-chip (SoC) implementation. In this case the BLE Host and the BLE
+  Controller communicate directly through function calls and queues in RAM. The
+  Bluetooth specification does not specify how HCI is implemented in this
+  single-chip configuration and so how HCI commands, events and data flows between
+  the two can be implementation-specific. The Zephyr OS supports this
+  configuration on Nordic Semiconductor ICs, for those applications and designs
+  that require a small footprint and the lowest possible power consumption, since
+  everything runs on a single IC
+* **Dual-chip configuration**: Using two separate ICs,
+  one running the Application and the Host, and a second one with the Controller
+  and the Radio Hardware. This is sometimes also called a connectivity-chip
+  configuration. This configuration allows for a wider variety of combinations of
+  Hosts when using the Zephyr OS as a Controller. Since HCI ensures
+  interoperability among Host and Controller implementations, including of course
+  Zephyr’s very own BLE Host and Controller, users of the Zephyr Controller can
+  choose to use whatever Host running on any platform they prefer. For example,
+  the host can be the Linux BLE Host stack (BlueZ) running on any processor
+  capable of supporting Linux. The Host processor may also run Zephyr and the
+  Zephyr OS BLE Host.
```

Comment:
Might want to mention here the case of combining Zephyr host with non-Zephyr controller, e.g. the case with qemu or native posix using your PC's internal Bluetooth adapter.

Reply:
Added thanks